src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -362,6 +362,39 @@ static void uct_ib_mlx5_devx_mr_lru_cleanup(uct_ib_mlx5_md_t *md)
     kh_destroy_inplace(rkeys, &md->lru_rkeys.hash);
 }
 
+static ucs_status_t
+uct_ib_mlx5_devx_reg_ksm(uct_ib_mlx5_md_t *md, uct_ib_mlx5_devx_mem_t *memh,
+                         uct_ib_mr_type_t mr_type, uint32_t iova_offset,
+                         int atomic, uint32_t mkey_index, const char *reason,
+                         struct mlx5dv_devx_obj **mr_p, uint32_t *mkey)
+{
+    uct_ib_mlx5_devx_mr_t *mr = &memh->mrs[mr_type];
+    uint64_t iova             = (uint64_t)memh->address + iova_offset;
+    void *address             = uct_ib_mlx5_devx_memh_base_address(memh);
+    ucs_status_t status;
+    size_t length;
+
+    if (memh->super.flags & UCT_IB_MEM_MULTITHREADED) {
+        ucs_assertv_always(mr->ksm_data->length > iova_offset,
+                           "mr->ksm_data->length=%ld iova_offset=%u",
+                           mr->ksm_data->length, iova_offset);
+        length = mr->ksm_data->length - iova_offset;
```

Comment:
why?

Reply:
We set iova (which is `address + iova_offset`) to address field during atomic ksm creation. If I pass full length, then iova + length exceed the registered buffer size and test fails during `devx_create_obj()` call.

@Artemy-Mellanox WDYT is that OK to pass `mr->ksm_data->length - iova_offset` as length?

Reply:
seems we were not doing it previously on `uct_ib_mlx5_devx_reg_atomic_key()` code path

Reply:
agree, seems it was mr->ksm_data->length previously, @ivankochin is this another bug fix?

Reply:
Yes, without the subtraction of the offset I get the following error: 
```
create mkey: window size exceed translations_octword_size
```

Reply:
why is it needed not after the change?